datafusion/functions/src/crypto/basic.rs

```diff
@@ -121,11 +121,9 @@ pub fn digest(args: &[ColumnarValue]) -> Result<ColumnarValue> {
         );
     }
     let digest_algorithm = match &args[1] {
-        ColumnarValue::Scalar(scalar) => match scalar {
-            ScalarValue::Utf8View(Some(method))
-            | ScalarValue::Utf8(Some(method))
-            | ScalarValue::LargeUtf8(Some(method)) => method.parse::<DigestAlgorithm>(),
-            other => exec_err!("Unsupported data type {other:?} for function digest"),
+        ColumnarValue::Scalar(scalar) => match scalar.try_as_str() {
```

Comment:
We could also avoid a bunch more duplication of stuff like this:

```
        let part = if let ColumnarValue::Scalar(ScalarValue::Utf8(Some(v))) = part {
```
If we added a similar convenience method `ColumnarValue::try_as_scalar_str()` that returned a Option<Option<&str>> 

Similarly we could do the same with `Expr::try_as_scalar_str()`